tests/rptest/tests/leadership_transfer_test.py

```diff
@@ -139,18 +139,40 @@ def all_partitions_present(nodes: int):
 
             return True
 
+        def count_leaders_per_node(topic_name: str):
+            leaders_per_node = collections.defaultdict(int)
+            tps = self.redpanda.partitions(topic_name)
+            for p in tps:
+                if p.leader:
+                    leaders_per_node[p.leader] += 1
+
+            return leaders_per_node
+
+        def distribution_error():
+            nodes = [
+                self.redpanda.node_id(n)
+                for n in self.redpanda.started_nodes()
+            ]
+            error = 0.0
+            for t in self.topics:
+                leaders_per_node = count_leaders_per_node(topic_name=t.name)
+                opt_leaders = t.partition_count / len(nodes)
+
+                for n in nodes:
+                    if n in leaders_per_node:
+                        error += (opt_leaders - leaders_per_node[n])**2
+                    else:
+                        error += opt_leaders**2
```

Comment:
I think the else part is not needed?

Reply:
we wouldn't account for the situation when there are no leaders on a node